events/ri/README.md

```diff
@@ -0,0 +1,137 @@
+# Nessie Events SPI - Reference Implementation
+
+This module contains reference implementations of Nessie's events notification system SPI module.
+
+## Overview
+
+Two implementations of the SPI module can be found in this module:
+
+1. `org.projectnessie.events.ri.console.PrintingEventSubscriber` - a (very) simple implementation 
+   that prints events to the console.
+2. `org.projectnessie.events.ri.kafka.KafkaEventSubscriber` - an implementation that publishes 
+   events to a Kafka topic.
+
+## PrintingEventSubscriber
+
+The `PrintingEventSubscriber` implementation is a very simple implementation that prints events to the console. It is
+intended to illustrate the basic concepts of an SPI implementation. Do NOT use in production.
+
+## KafkaEventSubscriber
+
+`KafkaEventSubscriber` illustrates how to handle events in a more realistic scenario. It is intended to be used as a
+starting point for implementing a real subscriber based on [Apache Kafka].
+
+[Apache Kafka]:https://kafka.apache.org/
+
+### Configuration
+
+The `KafkaEventSubscriber` implementation reads its configuration from a properties file. The default location of this
+file is `./nessie-kafka.properties`, but this can be changed by setting the `nessie.events.config.file` system property
+or the `NESSIE_EVENTS_CONFIG_FILE` environment variable to a different location.
+
+The properties file should contain all the desired Kafka producer configuration options, as well as the following
+subscriber-specific options:
+
+- `nessie.events.topic` - the name of the Kafka topic to publish events to. Defaults to `nessie-events`.
+- `nessie.events.repository-ids` - a comma-separated list of repository ids to watch. If not set, all repositories will
+  be watched. See below for more details.
+
+A template configuration file can be found in `src/main/resources/nessie-kafka.template.properties`.
+
+### Serialization and deserialization
+
+`KafkaEventSubscriber` uses the [Avro] library to define a domain model that is specific to the subscriber, and is not
+tied in any way to Nessie's. This is indeed the recommended approach, as it allows the subscriber to define its own
```

Comment:
I am not sure why tiding the implementation with Avro? Wouldn't make sense to have approach to let users definer their "choice" format? You could have some prefer using Avro, some may prefer protobuf and some may even prefer JSON :) . However, since is an example, could be fine though.

Reply:
I think this would go beyond the purpose of a reference implementation. As you said, this is an example and what matters most imho is that people can easily adapt it to their own needs. Here, it shouldn't be very hard to switch to protobuf for instance, so I don't see a lot of value in actually providing the protobuf bindings + another subscriber.